index.js

```diff
@@ -27,7 +39,75 @@ app.use(express.json());
 // IDEAS METHODS
 
 app.get('/ideas', (req, res) => {
-    res.status(200).json(getIdeas());
+    try {
+        if (req.query.sortBy === 'totalDonationSum') {
+            res.status(200).json(sortedByDonationSumIdeas());
+            return;
+        }
+        if (
+            req.query.sortBy === 'approve' &&
+            req.query.approve === 'accepted'
+        ) {
+            res.status(200).json(approvedIdeasList());
+            return;
+        }
+        if (req.query.sortBy === 'approve' && req.query.approve === 'pending') {
+            res.status(200).json(pendingIdeasList());
+            return;
+        }
+        if (
+            req.query.sortBy === 'approve' &&
+            req.query.approve === 'rejected'
+        ) {
+            res.status(200).json(rejectedIdeasList());
+            return;
+        }
+        res.status(200).json(getIdeas());
+    } catch (Error) {
```

Comment:
Why `Error` and not `error`?

Reply:
Because function throws Error, and I was thought that it is case sensitive, so wrote the same way which Error was  written there. now I've understood that it's the same error.

Reply:
Updated. Commited. Tested.

Reply:
There's a difference between `Error` and `error` in this case - try to figure out what's the difference.

Reply:
I've wrote it down, will take a look!